test/client.js

```diff
@@ -115,6 +115,11 @@ var operations = {
 describe('client', function () {
 
   var url = 'http://localhost:%s';
+  var hostIsNotReachableUrls =
+      {
+        ENOTFOUND: 'http://notthere:8088',
+        ECONNREFUSED: 'http://localhost:65535'
+      };
```

Comment:
you can use [nock#delayConnection](https://github.com/nock/nock#delay-the-connection) to test ETIMEDOUT

Reply:
Spent some time on trying to get that to work, however the hock API doesn't support this yet, see [here](https://github.com/mmalecki/hock/pull/34)

Reply:
Why hock vs. nock?

Reply:
I see we already have that in this library. huh. ok.